src/args.rs

```diff
@@ -199,18 +210,81 @@ impl Args {
         Ok(get_loaded_filter(self.filter.as_deref(), self.filter_arg.as_deref())?.map(Arc::new))
     }
 
-    pub(crate) fn get_loaded_query_text(&self) -> Result<String, Error> {
+    pub(crate) fn get_loaded_query_text_per_language(
+        &self,
+    ) -> Result<QueryOrQueryTextPerLanguage, Error> {
```

Comment:
This may seem like a weird "half-step" (toward `Args` just going ahead and actually parsing the query text), but I think it makes sense

The reason is that we do want to go ahead and fail fast if eg we got passed a path to a query file that we can't read

But eg for the case of auto-language (and not having passed `query_per_language`) we want to let `CachedQueries` lazily resolve the query text to a parsed query